lib/challenge.js

```diff
@@ -60,7 +82,14 @@ var some_pet;
 
 var Pet;
 // Answer Starts Here
-
+Pet = function Pet (name, species, breed, noise) {
+  this.name = name;
+  this.species = species;
+  this.breed = breed;
+  this.makeNoise = function(noise) {
+    console.log(noise);
+  }
+}
```

Comment:
In this case, since our Constructor function will be making many identical objects, we don't want it to make a brand new copy of our makeNoise method each time. To avoid this, we can use the prototype of Pet, and define all of our methods there; since all individual pets will have access to Pet.prototype, they can still each use all those methods, but it will use a lot less memory.

Syntactically, that would look like this:
```javascript
Pet = function(name, species, breed, noise){
  this.name = name;
  this.species = species;
  this.breed = breed;
  this.noise = noise;
}

Pet.prototype.makeNoise = function(){
  return this.noise;
};
```